src/NumericalAlgorithms/Interpolation/PredictedZeroCrossing.cpp

```diff
@@ -16,4 +17,31 @@ double predicted_zero_crossing_value(const std::vector<double>& x_values,
   return -coefficients[0]/coefficients[1];
 }
 
+DataVector predicted_zero_crossing_value(
+    const std::deque<double>& x_values,
+    const std::deque<DataVector>& y_values) {
+  ASSERT(x_values.size() == y_values[0].size(),
+         "The x_values and y_values must be of the same size");
+  intrp::LinearLeastSquares<1> predictor{x_values.size()};
+
+  DataVector result(y_values.size());
+
+  DataVector tmp_x_values{x_values.size()};
+  for (size_t i = 0; i < x_values.size(); i++) {
+    tmp_x_values[i] = x_values[i];
+  }
+
+  DataVector tmp_y_values{x_values.size()};
+  for (size_t i = 0; i < result.size(); i++) {
+    for (size_t j = 0; j < tmp_y_values.size(); j++) {
+      tmp_y_values[j] = y_values[i][j];
```

Comment:
Shouldn't this be `y_values[j][i]` ?    Each `i` indicates a point in the DataVector, and each `j` indicates a point in x_values.

Reply:
Yes, that's correct. I was not clear on how we needed to index through the y_values in my test which confused me when I was setting the tmp_y_values vector.